packages/bento-design-system/src/DateField/DateField.tsx

```diff
@@ -25,104 +23,125 @@ type SingleDateFieldProps = {
 type RangeDateFieldProps = {
   type: "range";
   shortcuts?: ShortcutProps<[Date, Date]>[];
-} & FieldProps<[Date | null, Date | null]>;
+} & FieldProps<[Date, Date] | null>;
 type Props = (SingleDateFieldProps | RangeDateFieldProps) & {
-  minDate?: UseDatepickerProps["minBookingDate"];
-  maxDate?: UseDatepickerProps["maxBookingDate"];
-  shouldDisableDate?: UseDatepickerProps["isDateBlocked"];
+  minDate?: Date;
+  maxDate?: Date;
+  shouldDisableDate?: (date: DateValue) => boolean;
   readOnly?: boolean;
 };
 
-export function DateField(props: Props) {
-  const inputConfig = useBentoConfig().input;
-  const startInputRef = useRef<HTMLInputElement>(null);
-  const endInputRef = useRef<HTMLInputElement>(null);
-  const [focusedInput, setFocusedInput] = useState<FocusedInput>(
-    props.autoFocus ? "startDate" : null
-  );
-  const [isFocused, setIsFocused] = useState(props.autoFocus ?? false);
-  const [isOpen, setIsOpen] = useState(props.autoFocus ?? false);
-  const validationState = props.readOnly ? "notSet" : props.issues ? "invalid" : "valid";
+function dateToCalendarDate(date: Date): CalendarDate {
+  if (!date) return date;
```

Comment:
I don't understand this line. In which case is `date` falsy and why is it ok to return a falsy value as `CalendarDate`?

Maybe we can use a more explicit check?

Reply:
leftover